test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala

```diff
@@ -153,6 +153,46 @@ class ExportsTest {
     assertEquals(43, foo.y)
   }
 
+  @Test def exports_for_abstract_properties_in_class_issue_2513(): Unit = {
+    abstract class Foo {
+      @JSExport
+      val x: Int
+      @JSExport
+      var y: Int
+    }
+
+    class Bar extends Foo {
+      val x: Int = 5
+      var y: Int = 6
+    }
+
+    val bar = (new Bar).asInstanceOf[js.Dynamic]
+    assertEquals(5, bar.x)
+    assertEquals(6, bar.y)
+    bar.y = 7
+    assertEquals(7, bar.y)
+  }
+
+  @Test def exports_for_abstract_properties_in_trait_issue_2513(): Unit = {
+    trait Foo {
+      @JSExport
+      val x: Int
+      @JSExport
+      var y: Int
+    }
+
+    class Bar extends Foo {
+      val x: Int = 5
+      var y: Int = 6
+    }
+
+    val bar = (new Bar).asInstanceOf[js.Dynamic]
+    assertEquals(5, bar.x)
+    assertEquals(6, bar.y)
+    bar.y = 7
```

Comment:
What do we expect when we do `bar.x = 8`? Does this call a setter that will throw?